### ISSUE TITLE
Heroic behaves differently running game (fail) vs running game with run exe on prefix (works)

### ISSUE THREAD

**Issue by stubkan** [Heroic-Games-Launcher/HeroicGamesLauncher#3459]
### Describe the bug

I have installed a prefix and a game (stalker gamma) and after a long while of being unable to run the game, I tried to run the game which was launched by a Mod Organizer 2 exe - by the "run exe on prefix" button and it worked.

So, it seems to me that Heroic operates somehow differently by running the same exe file on the same prefix fails the normal way, but succeeds the other way.  I cannot run the game normally, but am forced to "run exe on prefix" every time.

### Add logs

```text
Running normally Mod Organizer fails to start; by saying game files it needs are missing.  


Error 87 ERROR_INVALID_PARAMETER: Invalidparameter. (0x57)
 . binary: ''
 . owner: (file not found)
 . rights: (file not found)
 . arguments: ''
 . cwd: 'Z:\home\stubkan\DriveTwo\GAMMA'
 . stdout: no, stderr: no, hooked: yes
 . MO elevated: yes
 . usvfs x86:ok x64:ok proxy_x86:ok proxy_x64:ok
```

But when launching the same exe using 'run exe on prefix' button, it launches perfectly.

It may be, an issue with mounting folders - Mod Organizer does not like it when the game folder and the mod organizer folder are in the same folder, since it interferes with the virtual drive system that it uses to load mods - perhaps this is somehow occurring, even though they are not in the same folder?
```


### Steps to reproduce

Launch game containing mod organizer, fails

Launch same game with same prefix via 'run exe on prefix', works

### Expected behavior

Launch game should work, and not need to be run via exe via prefix

### Screenshots

_No response_

### Heroic Version

Latest Stable

### System Information

- OS - Ubuntu 22.04 Jammy (Mint 21)

### Additional information

I am unsure what information would be helpful, or where to find useful information - so will look forward to requests for logs or information when they say where to get it from.  Thanks in advance.

**Comment by stubkan** [comment]
@arielj 

But I cannot launch the game at all using the normal launcher, despite trying many times

I can only launch it if I run exe on prefix

I know that running exe on prefix is meant to be a one-off, yet this is the only way I can get this to run, therefore it is a bug

**Comment by arielj** [comment]
> @arielj
> 
> But I cannot launch the game at all using the normal launcher, despite trying many times
> 
> I can only launch it if I run exe on prefix
> 
> I know that running exe on prefix is meant to be a one-off, yet this is the only way I can get this to run, therefore it is a bug

please share a game log, it should show a lot of information (like the launch command, game settings, full output, etc)

**Comment by stubkan** [comment]
Sorry for the delay in response @arielj 

Here is the log when I launched normally (which causes Mod Organizer to crash and not start)
https://pastebin.com/pHdkCY30

Here is the log when I launched by selecting the same executable on the same prefix via 'run exe on prefix' (which causes Mod Organizer to start successfully)
https://pastebin.com/T00RjwU5

Both appear absolutely identical to me, so I am unsure if the log files update each runtime?  Anyway, this is why I am submitting this - since there is an unknown thing that Heroic is doing differently that is breaking it, and it does not seem to me the error lies with Mod Organizer - since the only variable changing here is Heroic Launcher.

diff says the difference is;
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/34361592/cfb3ac81-5426-4161-9144-6af66ed5fbab)

**Comment by arielj** [comment]
@flavioislima any idea why a sideloaded game would fail to launch with the `play` button but works with `run exe in prefix`? I know you are more familiar with that feature

**Comment by flavioislima** [comment]
> @flavioislima any idea why a sideloaded game would fail to launch with the `play` button but works with `run exe in prefix`? I know you are more familiar with that feature

Not really. I will look into it and try to replicate it.

**Comment by Qest333** [comment]
I got the same issue too and i think i know why it is happening.

When you launch the Mod Organizer with play now button, Heroic launches MO with additional arguments, even if you don't add any, in which case it passes an empty string that MO interprets as a name of the executable to launch alongside (shown as a binary variable), thus throwing an error. Launching MO with "Run EXE on prefix" doesn't pass any arguments so MO launches normally.

You can see how MO is launched from the mo_interface.log file in logs folder
Here's the log when it's launched incorrectly
```
[2024-07-28 12:44:42.030 D] command line: '"Z:\mnt\stuff\qest333\Heroic\Games\gamma\ModOrganizer.exe" ""'
[2024-07-28 12:44:42.030 I] starting Mod Organizer version 2.4.4 revision 1df1ea5e in Z:/mnt/stuff/qest333/Heroic/Games/gamma, usvfs: 0.5.6.0
[2024-07-28 12:44:42.030 I] data path: Z:/mnt/stuff/qest333/Heroic/Games/gamma
[2024-07-28 12:44:42.030 I] working directory: Z:/mnt/stuff/qest333/Heroic/Games/gamma
[2024-07-28 12:44:42.030 D] timing: MOApplication setup() 0 ms
[2024-07-28 12:44:42.408 I] using game plugin 'STALKER Anomaly' ('stalkeranomaly', variant (none), steam id '') at Z:/mnt/stuff/qest333/Heroic/Games/anomaly
[2024-07-28 12:44:42.539 W] "" not set up as executable
[2024-07-28 12:44:42.635 W] not saving lists during directory update
[2024-07-28 12:44:43.764 E] Error 87 ERROR_INVALID_PARAMETER: Niepoprawnyparametr. (0x57)
[2024-07-28 12:44:43.765 E]  . binary: ''
[2024-07-28 12:44:43.765 E]  . owner: (file not found)
[2024-07-28 12:44:43.765 E]  . rights: (file not found)
[2024-07-28 12:44:43.765 E]  . arguments: ''
[2024-07-28 12:44:43.765 E]  . cwd: 'Z:\mnt\stuff\qest333\Heroic\Games\gamma'
[2024-07-28 12:44:43.765 E]  . stdout: no, stderr: no, hooked: yes
[2024-07-28 12:44:43.765 E]  . MO elevated: yes
[2024-07-28 12:44:43.765 E]  . usvfs x86:ok x64:ok proxy_x86:ok proxy_x64:ok
```
And here's how it should look like
```
[2024-07-28 13:18:57.144 D] command line: '"Z:\mnt\stuff\qest333\Heroic\Games\gamma\ModOrganizer.exe"'
[2024-07-28 13:18:57.144 I] starting Mod Organizer version 2.4.4 revision 1df1ea5e in Z:/mnt/stuff/qest333/Heroic/Games/gamma, usvfs: 0.5.6.0
[2024-07-28 13:18:57.144 I] data path: Z:/mnt/stuff/qest333/Heroic/Games/gamma
[2024-07-28 13:18:57.144 I] working directory: Z:/mnt/stuff/qest333/Heroic/Games/gamma
[2024-07-28 13:18:57.144 D] timing: MOApplication setup() 0 ms
[2024-07-28 13:18:57.544 I] using game plugin 'STALKER Anomaly' ('stalkeranomaly', variant (none), steam id '') at Z:/mnt/stuff/qest333/Heroic/Games/anomaly
```

**Comment by Etaash-mathamsetty** [comment]
> I got the same issue too and i think i know why it is happening.
> 
> When you launch the Mod Organizer with play now button, Heroic launches MO with additional arguments (Game Arguments in Advanced section), even if you don't add any, in which case it passes an empty string that MO interprets as a name of the executable to launch alongside (shown as a binary variable), thus throwing an error. Launching MO with "Run EXE on prefix" doesn't pass any arguments so MO launches normally.
> 
> You can see how MO is launched from the mo_interface.log file in logs folder that is inside where MO is installed. Here's the log when it's launched via Play now button:
> 
> ```
> [2024-07-28 12:44:42.030 D] command line: '"Z:\mnt\stuff\qest333\Heroic\Games\gamma\ModOrganizer.exe" ""'
> [2024-07-28 12:44:42.030 I] starting Mod Organizer version 2.4.4 revision 1df1ea5e in Z:/mnt/stuff/qest333/Heroic/Games/gamma, usvfs: 0.5.6.0
> [2024-07-28 12:44:42.030 I] data path: Z:/mnt/stuff/qest333/Heroic/Games/gamma
> [2024-07-28 12:44:42.030 I] working directory: Z:/mnt/stuff/qest333/Heroic/Games/gamma
> [2024-07-28 12:44:42.030 D] timing: MOApplication setup() 0 ms
> [2024-07-28 12:44:42.408 I] using game plugin 'STALKER Anomaly' ('stalkeranomaly', variant (none), steam id '') at Z:/mnt/stuff/qest333/Heroic/Games/anomaly
> [2024-07-28 12:44:42.539 W] "" not set up as executable
> [2024-07-28 12:44:42.635 W] not saving lists during directory update
> [2024-07-28 12:44:43.764 E] Error 87 ERROR_INVALID_PARAMETER: Niepoprawnyparametr. (0x57)
> [2024-07-28 12:44:43.765 E]  . binary: ''
> [2024-07-28 12:44:43.765 E]  . owner: (file not found)
> [2024-07-28 12:44:43.765 E]  . rights: (file not found)
> [2024-07-28 12:44:43.765 E]  . arguments: ''
> [2024-07-28 12:44:43.765 E]  . cwd: 'Z:\mnt\stuff\qest333\Heroic\Games\gamma'
> [2024-07-28 12:44:43.765 E]  . stdout: no, stderr: no, hooked: yes
> [2024-07-28 12:44:43.765 E]  . MO elevated: yes
> [2024-07-28 12:44:43.765 E]  . usvfs x86:ok x64:ok proxy_x86:ok proxy_x64:ok
> ```
> 
> And here's the log when i use Run EXE on prefix:
> 
> ```
> [2024-07-28 13:18:57.144 D] command line: '"Z:\mnt\stuff\qest333\Heroic\Games\gamma\ModOrganizer.exe"'
> [2024-07-28 13:18:57.144 I] starting Mod Organizer version 2.4.4 revision 1df1ea5e in Z:/mnt/stuff/qest333/Heroic/Games/gamma, usvfs: 0.5.6.0
> [2024-07-28 13:18:57.144 I] data path: Z:/mnt/stuff/qest333/Heroic/Games/gamma
> [2024-07-28 13:18:57.144 I] working directory: Z:/mnt/stuff/qest333/Heroic/Games/gamma
> [2024-07-28 13:18:57.144 D] timing: MOApplication setup() 0 ms
> [2024-07-28 13:18:57.544 I] using game plugin 'STALKER Anomaly' ('stalkeranomaly', variant (none), steam id '') at Z:/mnt/stuff/qest333/Heroic/Games/anomaly
> ```

good find, will look at this

**Comment by arielj** [comment]
this should be fixed in 2.15.0 https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/3009
